File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;

    int i = 0;
    while (i < progresses.size()) {
        int progress = progresses[i], speed = speeds[i];
        int days = (100 - progress) / speed;
        if (progress + days * speed < 100) days++;

        int count = 1, j = i + 1;
        while (j < progresses.size()) {
            if (progresses[j] + days * speeds[j] < 100) break;
            count++;
            j++;
        }
        answer.emplace_back(count);
        i = j;
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var results: [Int] = []
    var answer: [Int] = [] 
    
    results = zip(progresses, speeds).map { (p, s) in 
        if (100-p) % s == 0 {
            return (100-p) / s
        } else {
            return (100-p) / s + 1
        }
    }
    
    var idx1: Int = 0, idx2 = idx1 + 1
    while idx1 < results.count {
        while idx2 < results.count && results[idx1] >= results[idx2] {
            idx2 += 1
        }
        answer.append(idx2 - idx1)
        idx1 = idx2
    }
    return answer
}